mobile/src/navigation/app/app.navigation.tsx

```diff
@@ -2,37 +2,61 @@ import { createDrawerNavigator } from '@react-navigation/drawer';
 import React, { FC } from 'react';
 
 import { AppScreenName } from '~/common/enums/enums';
-import { AppNavigationParamList } from '~/common/types/types';
-import { Billing } from '~/components/billing/billing';
-import { Courses } from '~/components/courses/courses';
-import { Mentors } from '~/components/mentors/mentors';
-import { MyEducation } from '~/components/my-education/my-education';
-import { Overview } from '~/components/overview/overview';
-import { Settings } from '~/components/setting/setting';
-import { UAM } from '~/components/uam/uam';
-
-import { SCREEN_OPTIONS } from './common/constants';
+import {
+  AppNavigationParamList,
+  DrawerNavigationItem,
+  PermissionItem,
+} from '~/common/types/types';
+import { useAppSelector, useMemo } from '~/hooks/hooks';
+
+import { NAVIGATION_ITEMS, SCREEN_OPTIONS } from './common/constants';
 import { DrawerContent } from './components/components';
 
 const Drawer = createDrawerNavigator<AppNavigationParamList>();
 
 const App: FC = () => {
+  const permissions = useAppSelector((state) => state.auth.user?.permissions);
+
+  const getAllowedScreens = (
+    screens: DrawerNavigationItem[],
+    userPermissions: PermissionItem[] | undefined,
+  ): DrawerNavigationItem[] => {
+    return screens.filter((screen) => {
+      if (screen.permissions.length === 0) return true;
+
+      return screen.permissions.some((permission) => {
+        return (
+          userPermissions?.filter(
+            (userPermission) => userPermission.id === permission.id,
+          ).length ?? false
+        );
+      });
```

Comment:
```suggestion
      return screen.permissions.every((permission) => {
        return (
          userPermissions?.some(
            (userPermission) => userPermission.id === permission.id,
          ),
        );
      });
```

Reply:
I think it will be better for the case where the screen has more than 1 permission

Reply:
it doesn't work for me.
![Screenshot from 2022-08-16 09-29-56](https://user-images.githubusercontent.com/12743599/184812606-205b51b8-3e86-4992-8674-5525c36044a0.png)

I expect to see the Courses item, but it's not there

![Screenshot from 2022-08-16 09-17-22](https://user-images.githubusercontent.com/12743599/184812862-0ebc189f-a07c-4266-81d4-ff3f864593d3.png)